processor/src/trace/virtual_bus/sum_check/mod.rs

```diff
@@ -0,0 +1,242 @@
+use super::univariate::UnivariatePolyCoef;
+use alloc::vec::Vec;
+use vm_core::FieldElement;
+
+mod prover;
+pub use prover::{FinalClaimBuilder, SumCheckProver};
+mod verifier;
+pub use verifier::{CompositionPolyQueryBuilder, SumCheckVerifier};
+
+/// A sum-check round proof.
+///
+/// This represents the partial polynomial sent by the Prover during one of the rounds of the
+/// sum-check protocol. The polynomial is in coefficient form and excludes the coefficient for
+/// the linear term as the Verifier can recover it from the other coefficients and the current
+/// (reduced) claim.
+#[derive(Debug, Clone)]
+pub struct RoundProof<E: FieldElement> {
+    pub round_poly_coefs: UnivariatePolyCoef<E>,
+}
```

Comment:
It seems like `RoundProof` and `UnivariatePolyCoef` are basically the same thing and are always used together.  I'm wondering if we need two separate structs for it. Alternatives could be:

1. Keep `RoundProof` and get rid of `UnivariatePolyCoef`. To do this, we'll need to move `evaluate_using_claim()` method into the `RoundProof` struct.
2. Keep `UnivariatePolyCoef` and get rid of `RoundProof` - this would take even fewer changes than option 1 above.

Reply:
Another option (especially if we go with the changes to the `sumcheck_round()` function described in https://github.com/0xPolygonMiden/miden-vm/pull/1307#discussion_r1612507007) could be to get rid of `RoundProof` and `UnivariatePolyEvals` and keep just `UnivariatePolyCoef` (and maybe rename it to just `UnivariatePoly`). This struct then could look like this:
```Rust
pub struct UnivariatePoly<E> {
    coefficients: Vec<E>,
}

impl<E: FieldElement> UnivariatePoly<E> {

    pub fn from_evaluations(evaluations: Vec<E>, round_claim: E) -> Self {
        // this would be basically the same method as UnivariatePolyEvals::to_poly()
    }

    pub fn evaluate_using_claim(&self, claim: E, challenge: E) -> E {
       // this method would be the same as now
    }
}
```